canvas/services/event.py

```diff
@@ -32,11 +34,12 @@ def _get_error_messages(submissions):
     return error_messages
 
 
-def _get_submission_status(submissions):
+def _get_submission_status(submissions, num_students):
     return {
         "Correct": submissions.filter(is_correct=True).count(),
         "Partially Correct": submissions.filter(is_correct=False, is_partially_correct=True).count(),
         "Incorrect": submissions.filter(is_correct=False, is_partially_correct=False).count(),
+        "Not Attempted": num_students - submissions.count(),
```

Comment:
This doesn't seem right. You are subtracting `number of submissions` from `number of students`. You can't subtract 2 numbers with different units. The only circumstance in which this makes sense is that the submissions are guaranteed to be the last submission of each student. Even if that's the case it's not obvious that this is a hidden assumption of this function.